### PR TITLE
ImportC: add tag symbol section

### PR DESCRIPTION
**Pull request by WalterBright**

### CODE REVIEW

File: spec/importc.dd [hunk 1]
```diff
@@ -143,6 +143,45 @@ $(H2 $(LNAME2 d-side, ImportC from D's Point of View))
...
+    int S;
+    struct S { int a, b; };
+    )
+
+    $(P the most pragmatic workaround is to add a typedef to the C code:)
+
+    $(CCODE
+    int S;
+    struct S { int a, b; };
+    typedef struct S S_t;    // add this typedef
```
Comment (ibuclaw) [inline spec/importc.dd:180]: I'm getting dejavu with another PR here.

Comment (WalterBright) [reply spec/importc.dd:180]: The other PR is about doing this as a wrapper.

File: spec/importc.dd [hunk 2]
```diff
@@ -170,6 +170,46 @@ $(H2 $(LNAME2 d-side, ImportC from D's Point of View))
...
+
+    $(CCODE
+    int S;
+    struct S { int a, b; };
+    S = 3;
+    struct S *ps;
+    )
+
+    $(P D does not have this distinction. If the only declaration of an identifier is the tag
+    symbol, that's what it sees. If there is a tag symbol and a non-tag symbol with the same
```
Comment (mdparker) [inline spec/importc.dd:187]: "that's what it sees" -> that's what the D compiler recognizes.  

"If there is a tag symbol and a non-tag symbol with the same identifier" -> Given a tag symbol and a non-tag symbol that share an identifier,

File: spec/importc.dd [hunk 3]
```diff
@@ -170,6 +170,46 @@ $(H2 $(LNAME2 d-side, ImportC from D's Point of View))
...
+    identifier, it sees the non-tag symbol. This is normally not a problem, due to the common
```
Comment (mdparker) [inline spec/importc.dd:188]: "it sees the non-tag symbol" -> the D compiler recognizes the non-tag symbol.

"a problem, due" -> a problem due

File: spec/importc.dd [hunk 4]
```diff
@@ -170,6 +170,46 @@ $(H2 $(LNAME2 d-side, ImportC from D's Point of View))
...
+    C practice of:)
```
Comment (mdparker) [inline spec/importc.dd:189]: "of:" -> of applying `typedef`, as in:

File: spec/importc.dd [hunk 5]
```diff
@@ -170,6 +170,46 @@ $(H2 $(LNAME2 d-side, ImportC from D's Point of View))
...
+
+    $(CCODE
+    typedef struct S { int a, b; } S;
+    )
+
+    $(P and D sees the typedef for `S` and it works as expected. But in the case of:)
```
Comment (mdparker) [inline spec/importc.dd:195]: "and D sees the typedef for `S` and it works" -> The D compiler recognizes the `typedef` applied to `S`, and the code compiles

"But in the case of" -> "But when `typedef` is absent, as in:

File: spec/importc.dd [hunk 6]
```diff
@@ -170,6 +170,46 @@ $(H2 $(LNAME2 d-side, ImportC from D's Point of View))
...
+
+    $(CCODE
+    int S;
+    struct S { int a, b; };
+    )
+
+    $(P the most pragmatic workaround is to add a typedef to the C code:)
```
Comment (mdparker) [inline spec/importc.dd:202]: "typedef" -> `typedef`

File: spec/importc.dd [hunk 7]
```diff
@@ -170,6 +170,46 @@ $(H2 $(LNAME2 d-side, ImportC from D's Point of View))
...
+
+    $(CCODE
+    int S;
+    struct S { int a, b; };
+    typedef struct S S_t;    // add this typedef
+    )
+
+    $(P and then D can access the struct tag symbol via `S_t`.)
```
Comment (mdparker) [inline spec/importc.dd:210]: "and then D" -> Then the D compiler

**Review by mdparker — CHANGES_REQUESTED** [review]
Some suggestions.

**Comment by dkorpel** [comment]
@WalterBright Can you incorporate Mike's suggestions?

**Review by dkorpel — APPROVED** [review]

**Pull request merged**